pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -295,3 +325,47 @@ func (c *Reconciler) reconcileVirtualService(ctx context.Context, ci *v1alpha1.C
 
 	return nil
 }
+
+func (c *Reconciler) reconcileGateways(ctx context.Context, ci *v1alpha1.ClusterIngress, gatewayNames []string) error {
+	for _, gatewayName := range gatewayNames {
+		if err := c.reconcileGateway(ctx, ci, gatewayName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Reconciler) reconcileGateway(ctx context.Context, ci *v1alpha1.ClusterIngress, gatewayName string) error {
+	// TODO(zhiminx): Need to handle the scenario when deleting ClusterIngress.
```

Comment:
Does this mean: We need to handle removing our contribution to the Gateway configuration when deleting the ClusterIngress?